src/Agent.Listener/Diagnostics/FolderPermissionInfo.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.IO;
+using System.Security.AccessControl;
+using System.Security.Principal;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener.Diagnostics
+{
+    class FolderPermissionInfo : IDiagnosticInfo
+    {
+        public void Execute(ITerminal terminal)
+        {
+            try
+            {
+                terminal.WriteLine("Checking for Read & Write Permissions");
+
+                string currentDirName = Directory.GetCurrentDirectory();
+                DirectoryInfo directoryInfo = new DirectoryInfo(currentDirName);
+                terminal.WriteLine($"{ directoryInfo.FullName.PadRight(75)} {HasFolderWritePermission(terminal, currentDirName)}");
```

Comment:
should this be a constant some place so that multiple diagnostic checks can rely on this rather than each one doing its own thing?